quinn-proto/src/connection.rs

```diff
@@ -323,6 +329,7 @@ impl Connection {
             data_recvd: 0,
             local_max_data: config.receive_window as u64,
             server_name: None,
+            tls_config: None,
```

Comment:
I don't like this very much. We're putting a `Config` state in here, optionally only for the initiators, which is only used in the very short time between the instantiation of the `Connection` and the `connect()`. Seems to me like we might want to create a `new_client()` constructor instead that just does the `connect()` bits straightaway, and then we don't have to store the `ClientConfig`.

Reply:
The `ClientConfig` is stored for use in resetting state on retry packet receipt. That said, I'm not actually sure resetting TLS state is necessary; tests pass and draft 11 doesn't seem to specifically require it. It won't be needed in draft 15 either, so I'll just strip it out.